src/drivers/can.rs

```diff
@@ -0,0 +1,1502 @@
+    D0 = 0x6,
+    /// TXBND1/RXBND1 - Data byte 1
+    D1 = 0x7,
+    /// TXBND2/RXBND2 - Data byte 2
+    D2 = 0x8,
+    /// TXBND3/RXBND3 - Data byte 3
+    D3 = 0x9,
+    /// TXBND4/RXBND4 - Data byte 4
+    D4 = 0x10,
+    /// TXBND5/RXBND5 - Data byte 5
+    D5 = 0x11,
+    /// TXBND6/RXBND6 - Data byte 6
+    D6 = 0x12,
+    /// TXBND7/RXBND7 - Data byte 7
+    D7 = 0x13,
+}
+
+/// MCP2515 CANCTRL register setting bits.  
+#[derive(Clone, Copy)]
+pub struct SettingsCanCtrl {
+    mode: OperationTypes,
+    clken: bool,
+    clkpre: CLKPRE,
+    abat: bool, // ABAT: Abort All Pending Transmission bit
+    osm: bool,  // OSM: One-Shot Mode (1 = Enabled, 0 = disabled)
+}
+
+/// MCP2515 register CANINTE.
+#[repr(u8)]
+#[derive(Clone, Copy)]
+#[allow(clippy::module_name_repetitions)]
+pub enum CanInte {
+    /// Message Error Interrupt bit - during message reception/transmission.
+    MERRE = 0b1000_0000,
+
+    /// Wake-up Interrupt bit - on CAN bus activity.
+    WAKIE = 0b0100_0000,
+
+    /// Error Interrupt bit - on EFLG error condition change.
+    ERRIE = 0b0010_0000,
+
+    /// Transmit buffer 2 empty interrupt bit - when TXB2 becomes empty (sent
+    /// message).
+    TX2IE = 0b0001_0000,
+
+    /// Transmit buffer 1 empty interrupt bit - when TXB1 becomes empty (sent
+    /// message).
+    TX1IE = 0b0000_1000,
+
+    /// Transmit buffer 0 empty interrupt bit - when TXB0 becomes empty (sent
+    /// message).
+    TX0IE = 0b0000_0100,
+
+    /// Receive buffer 1 full interrupt bit - message received in RXB1.
+    RX1IE = 0b0000_0010,
+
+    /// Receive buffer 0 full interrupt bit - message received in RXB0.
+    RX0IE = 0b0000_0001,
+}
+
+/// MCP2515 register CANSTAT.ICOD - Interrupt Flag Code bits.
+#[repr(u8)]
+#[derive(Debug, Format, Clone, Copy)]
+pub enum InterruptFlagCode {
+    NoInterrupt = 0b000,
+    ErrorInterrupt = 0b001,
+    WakeUpInterrupt = 0b010,
+    TXB0Interrupt = 0b011,
+    TXB1Interrupt = 0b100,
+    TXB2Interrupt = 0b101,
+    RXB0Interrupt = 0b110,
+    RXB1Interrupt = 0b111,
+}
+
+/// MCP2515 Clock speed in Mega-hertz.
+#[derive(Clone, Copy)]
+pub enum McpClock {
+    /// 16 `MHz`.
+    MCP16,
+
+    /// 8 `MHz`.
+    MCP8,
+}
+
+/// Transmission rate in kBPS.
+#[derive(Clone, Copy)]
+pub enum Bitrate {
+    /// 125 kBPS
+    CAN125,
+}
+
+#[derive(Clone, Copy)]
+/// Acceptable frame ID that the RX0 or RX1 buffer should accept on the CAN bus.
+/// This would determine if a message should be loaded into either of the
+/// receive buffers.
+///
+/// - `rx_mask` - the mask determine which bits in the identifier that should be
+///   examined with the filter.
+///
+/// - `acceptance filter` - are compared with identifier fields of the message.
+pub struct AcceptanceFilterMask {
+    rx_mask: u16,
+    acceptance_filter: u16,
+}
+
+/// Struct for neccessary settings for setting up the MCP2515 CAN module.
+#[derive(Clone, Copy)]
+pub struct Mcp2515Settings {
+    pub canctrl: SettingsCanCtrl,
```

Comment:
Either these fields should not be public or they should have doc comments

Reply:
Changed to private, and added a "new" method for creating a new Mcp2515Settings instance.